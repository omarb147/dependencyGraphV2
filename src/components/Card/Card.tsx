import React from 'react';
import styled from 'styled-components';

interface ICardProps {
  color: string;
  text: string;
  size: {
    height: number;
    width: number;
  };
  selected?: boolean;
  className?: string;
}

const Card: React.SFC<ICardProps> = ({ className, text }: ICardProps) => (
  <div className={className}>
    <div>{text}</div>
  </div>
);

export default styled(Card)`
  background-color: ${(props): string => props.color};
  width: 100%;
  height: 100%;
  div {
    font-size: ${(props): number => props.size.width / 10 + 8}px;
  }
`;
