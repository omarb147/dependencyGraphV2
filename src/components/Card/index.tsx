import React from 'react';
import styled from 'styled-components';
import NodesSelectorClass from '@/module/node/selectors';

interface ICardFrameProps {
  color: string;
  selected: boolean;
  className?: string;
}

interface ICardProps {
  selected: boolean;
  itemId: string;
  className?: string;
}

const CardFrame = styled.div<ICardFrameProps>`
  background-color: #ffffff;
  border: ${(props): string => (props.selected ? `5px solid ${props.color}` : 'none')};
  border-top: 5px solid ${(props): string => props.color};
  width: 20rem;
  box-shadow: 0px 5px 10px #c9c9c9;
  padding: 0.75rem;
`;


const UserStoryWrapper = styled.div`
  color: #434343;
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
`;

const Point = styled.div<{ color: string }>`
  padding: 0.25rem 0.4rem;
  font-size: 0.8rem;
  background-color: ${(props): string => props.color};
  color: #ffffff;
  margin-right: 0.5rem;
  opacity: 0.7;
`;

const Divider = styled.hr`
  margin: 0.5rem 0 0.25rem;
  border-top: 1px solid #e1e1e1;
`;

const Text = styled.div`
  font-size: 0.9rem;
  flex-grow: 1;
`;

const LabelFrame = styled.div`
  display: flex;
  justify-content: flex-end;
  color: #a5a5a5;
  font-size: 0.7rem;
`;

const Card: React.SFC<ICardProps> = ({
  className,
  itemId,
  selected
}: ICardProps) => {
  const NodesSelector = new NodesSelectorClass();
  const {
    points,
    color,
    labels,
    name,
    status,
  } = NodesSelector.useTicketById(itemId);

  return (
    <CardFrame
      color={color}
      className={className}
      selected={selected}
    >
      <UserStoryWrapper>
        {points ? (
          <Point color={color}>
            {points}
          </Point>
        ) : null}
        <Text>{name}</Text>
      </UserStoryWrapper>
      <Divider />
      {labels ? (
        <>
          <LabelFrame>{labels}</LabelFrame>
        </>
      ) : null}
    </CardFrame>
  );
};

export default Card;
