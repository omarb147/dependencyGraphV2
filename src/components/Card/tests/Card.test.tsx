import React from 'react';
import { render } from '@testing-library/react';
import Card from '../Card';

describe('<Card/>', () => {
  it('should render a card with color and text', () => {
    const size = { width: 100, height: 200 };
    const { getByText } = render(<Card text="Hello World" color="yellow" size={size} />);
    expect(getByText('Hello World')).toMatchSnapshot();
  });
});
