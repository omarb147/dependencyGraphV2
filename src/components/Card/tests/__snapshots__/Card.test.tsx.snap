// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> should render a card with color and text 1`] = `
<div>
  Hello World
</div>
`;
